src/server/internalServer.cpp

```diff
@@ -802,8 +810,15 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
     if (m_verbose.load())
       printf("Failed to find %s\n", urlStr.c_str());
 
-    return Response::build_404(*this, request, bookName);
+    MustacheData data;
+    data.set("root", m_root);
+    data.set("bookName", bookName);
+    data.set("pattern", pattern);
+    kainjow::mustache::mustache tmpl("Make a full text search for <a href=\"{{{root}}}/search?content={{{bookName}}}&pattern={{{pattern}}}\">{{pattern}}</a>");
```

Comment:
At this point `pattern` is URL-encoded (since it was extracted as a fragment of a URL). When generating the HTML, pattern must stay URL-encoded in the search URL, but must be URL-decoded as text to be presented to the user (and we let mustache HTML-encode it).

The best solution may be to compose the search URL via C++ and pass it as a single parameter to mustache. Then you can create and share a mustache template with the other fragment of related code.